akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/SnapshotTestKit.scala

```diff
@@ -0,0 +1,305 @@
+
+  /**
+   * Fail next read from storage (recovery) attempt with default exception for any persistence id.
+   */
+  def failNextRead(persistenceId: String): Unit = failNextNReads(persistenceId, 1)
+
+  /**
+   * Fail next n read from storage (recovery) attempts with `cause` exception for any persistence id.
+   */
+  def failNextNReads(n: Int, cause: Throwable): Unit = scalaTestkit.failNextNReads(n, cause)
+
+  /**
+   * Fail next n read from storage (recovery) attempts with default exception for any persistence id.
+   */
+  def failNextNReads(n: Int): Unit = failNextNReads(n, ExpectedFailure)
+
+  /**
+   * Fail next n read from storage (recovery) attempts with `cause` exception for particular persistence id.
+   */
+  def failNextNReads(persistenceId: String, n: Int, cause: Throwable): Unit =
+    scalaTestkit.failNextNReads(persistenceId, n, cause)
+
+  /**
+   * Fail next n read from storage (recovery) attempts with default exception for particular persistence id.
+   */
+  def failNextNReads(persistenceId: String, n: Int): Unit = failNextNReads(persistenceId, n, ExpectedFailure)
+
+  /**
+   * Fail next delete from storage attempt with `cause` exception for any persistence id.
+   */
+  def failNextDelete(cause: Throwable): Unit = failNextNDeletes(1, cause)
+
+  /**
+   * Fail next delete from storage attempt with default exception for any persistence id.
+   */
+  def failNextDelete(): Unit = failNextNDeletes(1)
+
+  /**
+   * Fail next delete from storage attempt with `cause` exception for particular persistence id.
+   */
+  def failNextDelete(persistenceId: String, cause: Throwable): Unit = failNextNDeletes(persistenceId, 1, cause)
+
+  /**
+   * Fail next delete from storage attempt with default exception for particular persistence id.
+   */
+  def failNextDelete(persistenceId: String): Unit = failNextNDeletes(persistenceId, 1)
+
+  /**
+   * Fail next n delete from storage attempts with `cause` exception for any persistence id.
+   */
+  def failNextNDeletes(n: Int, cause: Throwable): Unit = scalaTestkit.failNextNDeletes(n, cause)
+
+  /**
+   * Fail next n delete from storage attempts with default exception for any persistence id.
+   */
+  def failNextNDeletes(n: Int): Unit = failNextNDeletes(n, ExpectedFailure)
+
+  /**
+   * Fail next n delete from storage attempts with `cause` exception for particular persistence id.
+   */
+  def failNextNDeletes(persistenceId: String, n: Int, cause: Throwable): Unit =
+    scalaTestkit.failNextNDeletes(persistenceId, n, cause)
+
+  /**
+   * Fail next n delete from storage attempts with default exception for particular persistence id.
+   */
+  def failNextNDeletes(persistenceId: String, n: Int): Unit = failNextNDeletes(persistenceId, n, ExpectedFailure)
+
+  /**
+   * Check that `msgs` events have been persisted in the storage in order.
+   */
+  def expectPersistedInOrder[A](persistenceId: String, msgs: JList[A]): JList[A] =
```

Comment:
a simplification for all these that assert several events would be to replace them with one single `receivePersisted(n: Int)` and the user can do the actual asserts using ScalaTest/JUnit.

See `receiveMessages` in `akka.actor.testkit.typed.scaladsl.TestProbe`